src/encoded/static/components/facets/defaults.js

```diff
@@ -940,11 +948,54 @@ export const DefaultFacet = ({ facet, results, mode, relevantFilters, pathname,
     // This should ultimately be accomplished in the back end, but the front end fix is much simpler so we are starting with that
     // We have to check the full list for now (until schema change) because some lists contain both numerical and string terms ('Encyclopedia version' under Annotations) and we do not want to sort those by value
     const numericalTest = a => !isNaN(a.key);
+
     // For straightforward numerical facets, just sort by value
     const processedTerms = significantTerms.every(numericalTest) ? _.sortBy(significantTerms, obj => obj.key) : significantTerms;
 
     const disabledCss = filters => filters.some(f => f.field.indexOf('!') !== -1 && f.term.trim() === '*');
 
+    React.useEffect(() => {
+        const field = facet.field;
+
+        // Get facets from storage that need to be expanded
+        const facetsStorage = sessionStorage.getItem(FACET_STORAGE);
+        const facetList = new Set(facetsStorage ? facetsStorage.split(',') : []);
+
+        // This determines if facet has just loaded check if anything needs
+        // to be expanded
+        const newlyLoadedFacetStorage = `${MARKER_FOR_NEWLY_LOADED_FACET_PREFIX}${field}`;
+        const storedFacetNewlyLoaded = sessionStorage.getItem(newlyLoadedFacetStorage);
+
+        if (!storedFacetNewlyLoaded) {
+            // sessions storage set to prevent this if-statement from running again on the field
+            sessionStorage.setItem(newlyLoadedFacetStorage, field);
+
+            // auto-open facets based on (1)With selected terms OR facet pre-determined to be opened
+            if ((relevantFilters && relevantFilters.length > 0) || preopenedFacetList.includes(field)) {
+                facetList.add(field);
+            }
+        }
+
+        sessionStorage.setItem(FACET_STORAGE, facetList.size !== 0 ? [...facetList].join(',') : []);
+        setExpandFacets(facetList); // initalize facet collapse-state
+    }, []); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
I feel like this code should be handled by search.js, not the default facet. If we wanted to make the date-selector facet or the exists facet also collapse (and I felt surprised they don’t), then we’ll have to copy this code to the other kinds of default and custom facets. I think each facet component should be told through props if they appear expanded or not. We’ll still need a trigger callback, but search.js should be the one and only place that stores the expanded/collapsed states. The facet and title components should only know whether their particular facet appears expanded or not.